src/app/Components/email-preview/email-preview.component.html

```diff
@@ -18,7 +18,12 @@
       </tr>
     </thead>
     <tbody>
-      <tr class="cursor-pointer" *ngFor="let data of dataPreview">
+      <tr
+        class="cursor-pointer"
+        *ngFor="let data of dataPreview"
+        (click)="openEmailDetail(data)"
```

Comment:
send the id of the item instead of the object itself: `openEmailDetail(data?._id)`